weatherAplication/Modules/Home/View/HomeViewController.swift

```diff
@@ -25,6 +27,12 @@ class HomeViewController: UIViewController {
         fatalError("init(coder:) has not been implemented")
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        viewModel.refresh()
+        lastCityNamesTable.reloadData()
```

Comment:
Super, właśnie o to chodziło w refreshowaniu tych danych. Ale jeszcze parę uwag do tego. Po pierwsze wywoływana nazwa funkcji jest zła, powinna ona przekazywać do viewModelu informację co się dzieje w viewControllerze, a nie co ma viewModel ma zrobić. O tym co powinno się stać decyduje viewModel i viewController się tym nie interesuje. Po drugie refreshowanie lastCityNamesTable powinno być wywołane z poziomu viewModelu- znowu przenosisz odpowiedzialność na viewController.